_source/_posts/2022-01-13-spring-cloud-contract.md

```diff
@@ -0,0 +1,893 @@
+   }
+
+   @EnableWebSecurity
+   public class OAuth2ResourceServerSecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+      @Override
+      protected void configure(HttpSecurity http) throws Exception {
+         http.authorizeRequests()
+             .anyRequest().authenticated()
+             .and()
+             .oauth2ResourceServer().jwt();
+      }
+   }
+}
+```
+
+Finally, you need to add the Issuer URI to your `application.properties` file. You need to replace `{yourOktaDomain}` in the line below with your Okta domain. 
+
+`src/main/resources/application.properties`
+
+```properties
+okta.oauth2.issuer=https://{yourOktaDomain}/oauth2/default
+```
+
+You can find your Okta URI by typing `okta login` from the command line. It should print out something like the following.
+
+```bash
+Okta Org already configured: https://dev-123456.okta.com/
+```
+
+You can also find your Issuer URI by opening your Okta Admin Console and navigating to **Security** > **API** and looking at the **Issuer URI** for the `default` authorization server. 
+
+Start the application (or, if it is still running, use `control-c` to stop it and restart it).
+
+```bash
+./mvnw spring-boot:run
+```
+## Generate a JWT and Test the Secure App
+
+
+To generate a valid JWT, you need to create an OpenID Connect (OIDC) application on Okta and use the OIDC Debugger application to request a JWT from the OIDC application. 
+
+{% include setup/cli.md type="web" install="false" loginRedirectUri="https://oidcdebugger.com/debug" %}
+
+{% include setup/oidcdebugger.md %}
+
+Once you have the token value, save it in a shell variable (in the terminal window where you are going to run the HTTPie command to test the authenticated application).
+
+```bash
+TOKEN=eyJraWQiOiJDVkdzamc2MDBCNVFXR3VSdVRDckJuLVhLa2J...
+```
+
+Test the authenticated application using the token.
+
+```bash
+http :8080/hat/1 "Authorization: Bearer $TOKEN"
+```
+
+Make sure you use double quotes in the command above, otherwise the variable will not be expanded.
+
+You should get the first hat returned.
+
+```bash
+HTTP/1.1 200 
+...
+{
+    "color": "red",
+    "id": 1,
+    "name": "Sombrero",
+    "size": 30
+}
+```
+
+One of the benefits of the Spring Cloud Contract project is that the mocked and stubbed test objects bypass authentication. This is super helpful (as long as you are not testing application security, of course). You can run `./mvnw clean install` or `./mvnw test` and you will see that the integration tests still pass and that the stubs are still being generated and deployed.
+
+## Create the Consumer Application
+
+The next step is to create the consumer application. This will simulate a public-facing service that will retrieve hats from the producer and serve them to the clients. The key point in this tutorial is to see how the stubs generated from the producer project can be imported into the consumer project and used to verify application behavior.  
```

Comment:
```suggestion
The next step is to create the consumer application. This will simulate a public-facing service to retrieve hats from the producer and serve them to the clients. The key point in this tutorial is to see how the stubs generated from the producer project can be imported into the consumer project and used to verify application behavior.  
```